zookeeper-server/src/main/java/org/apache/zookeeper/common/ClientX509Util.java

```diff
@@ -35,5 +59,95 @@ protected boolean shouldVerifyClientHostname() {
     public String getSslAuthProviderProperty() {
         return sslAuthProviderProperty;
     }
+    /**
+     * Determine the string to be used as the remote host session Id for
+     * authorization purposes. Associate this client identifier with a
+     * ServerCnxn that has been authenticated over SSL, and any ACLs that refer
+     * to the authenticated client.
+     *
+     * @param clientCert Authenticated X509Certificate associated with the
+     *                   remote host.
+     * @return Identifier string to be associated with the client.
+     */
+    public String getClientId(X509Certificate clientCert) {
```

Comment:
Should we leave `getClientId()` where it was since it originally belonged to X509 Provider?

Reply:
As discussed offline, this method will be used by both X509AuthenticationProvider and X509ZNodeGroupAclProvider, so I moved it to Util class X509AuthenticationUtil. Let me know if it sounds reasonable.

Reply:
Great!